--- jquery-1.4.2.js	2010-02-14 04:35:05.000000000 +0100
+++ jquery-1.4.2.mod.js	2010-04-21 14:04:26.000000000 +0200
@@ -1,5 +1,5 @@
 /*!
- * jQuery JavaScript Library v1.4.2
+ * jQuery JavaScript Library v1.4.2svg
  * http://jquery.com/
  *
  * Copyright 2010, John Resig
@@ -1643,11 +1643,12 @@
 				// events handler returns false
 				if ( !special.setup || special.setup.call( elem, data, namespaces, eventHandle ) === false ) {
 					// Bind the global event handler to the element
-					if ( elem.addEventListener ) {
-						elem.addEventListener( type, eventHandle, false );
-
-					} else if ( elem.attachEvent ) {
-						elem.attachEvent( "on" + type, eventHandle );
+					try { // SVG
+						elem.addEventListener(type, eventHandle, false);
+					}
+					catch(e) {
+						if (elem.attachEvent)
+							elem.attachEvent("on" + type, eventHandle);
 					}
 				}
 			}
@@ -1769,7 +1770,13 @@
 			// remove generic event handler if no more handlers exist
 			if ( eventType.length === 0 || pos != null && eventType.length === 1 ) {
 				if ( !special.teardown || special.teardown.call( elem, namespaces ) === false ) {
-					removeEvent( elem, type, elemData.handle );
+					try { // SVG
+						elem.removeEventListener(type, jQuery.data(elem, "handle"), false);
+					}
+					catch (e) {
+						if (elem.detachEvent)
+							elem.detachEvent("on" + type, jQuery.data(elem, "handle"));
+					}
 				}
 
 				ret = null;
@@ -2744,13 +2751,13 @@
 		} else if ( context && context.nodeType === 1 ) {
 			for ( var i = 0; checkSet[i] != null; i++ ) {
 				if ( checkSet[i] && (checkSet[i] === true || checkSet[i].nodeType === 1 && contains(context, checkSet[i])) ) {
-					results.push( set[i] );
+					results.push( set[i] || set.item(i) ); // SVG
 				}
 			}
 		} else {
 			for ( var i = 0; checkSet[i] != null; i++ ) {
 				if ( checkSet[i] && checkSet[i].nodeType === 1 ) {
-					results.push( set[i] );
+					results.push( set[i] || set.item(i) ); // SVG
 				}
 			}
 		}
@@ -3348,7 +3355,7 @@
 		} else {
 			if ( typeof array.length === "number" ) {
 				for ( var i = 0, l = array.length; i < l; i++ ) {
-					ret.push( array[i] );
+					ret.push( array[i] || array.item(i) ); // SVG
 				}
 			} else {
 				for ( var i = 0; array[i]; i++ ) {
